tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses_test.py

```diff
@@ -0,0 +1,160 @@
+Step 3:
+ Keep only records associated with most questionnaire_response_id, person, question, answer per each cope_month version.
+"""
+
+# Python Imports
+import os
+
+# Project Imports
+from app_identity import PROJECT_ID
+from cdr_cleaner.cleaning_rules.drop_cope_duplicate_responses import DropCopeDuplicateResponses
+from tests.integration_tests.data_steward.cdr_cleaner.cleaning_rules.bigquery_tests_base import BaseTest
+
+
+class DropCopeDuplicateResponsesTest(BaseTest.CleaningRulesTestBase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+        super().initialize_class_vars()
+
+        # Set the test project identifier
+        project_id = os.environ.get(PROJECT_ID)
+        cls.project_id = project_id
+
+        # Set the expected test datasets
+        dataset_id = os.environ.get('RDR_DATASET_ID')
+        cls.dataset_id = dataset_id
+        sandbox_id = dataset_id + '_sandbox'
+        cls.sandbox_id = sandbox_id
+
+        cls.rule_instance = DropCopeDuplicateResponses(project_id, dataset_id,
+                                                       sandbox_id)
+
+        sb_table_names = cls.rule_instance.get_sandbox_tablenames()
+        for table_name in sb_table_names:
+            cls.fq_sandbox_table_names.append(
+                f'{project_id}.{sandbox_id}.{table_name}')
+
+        cls.fq_table_names = [f'{project_id}.{dataset_id}.observation']
+
+        # call super to set up the client, create datasets, and create
+        # empty test tables
+        # NOTE:  does not create empty sandbox tables.
+        super().setUpClass()
+
+    def setUp(self):
+        """
+        Create common information for tests.
+
+        Creates common expected parameter types from cleaned tables and a common
+        fully qualified (fq) dataset name string to load the data.
+        """
+        fq_dataset_name = self.fq_table_names[0].split('.')
+        self.fq_dataset_name = '.'.join(fq_dataset_name[:-1])
+
+        super().setUp()
+
+    def test_field_cleaning(self):
+        """
+        Tests that the specifications for the SANDBOX_QUERY and CLEAN_PPI_NUMERIC_FIELDS_QUERY
+        perform as designed.
+
+        Validates pre conditions, tests execution, and post conditions based on the load
+        statements and the tables_and_counts variable.
+        """
+
+        tmpl = self.jinja_env.from_string("""
+            INSERT INTO `{{fq_dataset_name}}.observation` (observation_id,
+    person_id,
+    observation_concept_id,
+    observation_date,
+    observation_datetime,
+    observation_type_concept_id,
+    observation_source_concept_id,
+    observation_source_value,
+    value_source_value,
+    questionnaire_response_id)
+VALUES
```

Comment:
Add the following test cases
- Exactly the same column values except for `observation_id`, we want to drop this
- Different `observation_id` and different `value_source_value`, keep everything else the same, we want to preserve this record.

Reply:
done added!